test/sync_test.go

```diff
@@ -152,36 +181,38 @@ func testSyncCluster(t *testing.T) {
 	expected := [][]fileInfo{e1, e2, e3}
 
 	// Start the syncers
-	p, err := scStartProcesses()
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer func() {
-		for i := range p {
-			p[i].stop()
-		}
-	}()
 
-	log.Println("Waiting for startup...")
-	for _, dev := range p {
-		waitForScan(dev)
-	}
+	log.Println("Starting Syncthing...")
+
+	p0 := startInstance(t, 1)
+	defer checkedStop(t, p0)
+	p1 := startInstance(t, 2)
+	defer checkedStop(t, p1)
+	p2 := startInstance(t, 3)
+	defer checkedStop(t, p2)
+
+	p := []*rc.Process{p0, p1, p2}
+
+	start := time.Now()
+	iteration := 0
+	for time.Since(start) < timeLimit {
+		iteration++
+		log.Println("Iteration", iteration)
 
-	for count := 0; count < iterations; count++ {
 		log.Println("Forcing rescan...")
 
 		// Force rescan of folders
 		for i, device := range p {
-			if err := device.rescan("default"); err != nil {
+			if err := device.Rescan("default"); err != nil {
 				t.Fatal(err)
 			}
-			if i < 2 {
-				if err := device.rescan("s12"); err != nil {
+			if i == 0 || i == 1 {
+				if err := device.Rescan("s12"); err != nil {
 					t.Fatal(err)
 				}
 			}
-			if i > 1 {
-				if err := device.rescan("s23"); err != nil {
+			if i == 1 || i == 2 {
+				if err := device.Rescan("s23"); err != nil {
```

Comment:
This should be RescanDelay because https://github.com/calmh/syncthing/blob/b79d32d9b8afcef3d26a6e0c539350fe574df2c3/test/sync_test.go#L274 could already contain changes.